security/secure-markup.md

```diff
@@ -0,0 +1,90 @@
+
+This may sound complicated but it's actually easy to do!
+
+Example markup (from the [MDN article on Subresource Integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity)):
+```
+<script src="https://example.com/example-framework.js"
+  integrity="sha384-oqVuAfXRKap7fdgcCY5uykM6+R9GqQ8K/uxy9rx7HNQlGYl1kPzQho1wx4JwY8wC"
+  crossorigin="anonymous"></script>
+```
+
+Use of the `crossorigin` attribute depends on [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) support of the asset server.
+
+Subresource Integrity can be used in conjunction with a [Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) but does not require a CSP to work.
+
+At the time of writing, [browser support for Subresource Integrity](https://caniuse.com/#feat=subresource-integrity) is very good and getting better.
+
+As Subresource Integrity should be relatively easy to add to a front-end build tool chain, it's recommended you use it for any `script` or `link` element.
+
+- [Gentle introduction to Subresource Integrity by keycdn.com](https://www.keycdn.com/support/subresource-integrity/)
+- [MDN article on Subresource Integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity)
+
+## Use sane form defaults
+
+We must not write data that comes from the user into the DOM without first [sanitising](https://www.smashingmagazine.com/2011/01/keeping-web-users-safe-by-sanitizing-input-data/) that data. Input includes not only form data, but also things like user agent strings, request headers, URL parameters and cookies.
+
+Sanitising input is the responsibility of the server in most cases, but we must also sanitise input in rich JavaScript applications (e.g. URL fragments) at the risk of exposing our users to attacks like [DOM XSS/Client-Side XSS](https://www.owasp.org/index.php/Types_of_Cross-Site_Scripting#DOM_Based_XSS_.28AKA_Type-0.29).
+
+Such sanitisation is beyond the scope of this document, but as authors of HTML forms we can mitigate _some_ classes of client-side attack by specifying some form widget attributes. This is no substitute for sanitisation but constitutes part of a ["belt & braces"](https://www.collinsdictionary.com/dictionary/english/belt-and-braces) approach to security, as well as reducing a common source of bugs.
+
+- `form` &mdash; specify `accept-charset="utf-8"`. Charset issues are a common source of bugs generally. If a page has been parsed as UTF-8 the browser _should_ default to sending data as UTF-8, but it's easy to specify the `accept-charset` as extra protection in case something breaks, so why not do it?
+- `input` &mdash; if the value of the type attribute is `text`, `email`, `search`, `password`, `tel`, or `url`, specify the `maxlength` attribute.
```

Comment:
Is it worth noting something on `inputmode` here? 
https://bugs.chromium.org/p/chromium/issues/detail?id=244688

Reply:
Well I learnt something today, didn't know about that new attr! Not sure if worth mentioning or not, what might the security implications be?

Reply:
Hmm, actually it's probably not applicable here. I was thinking about general markup guidelines, but this is about security. 

I just got carried away @jpw!